viewing_party/party.py

```diff
@@ -1,23 +1,276 @@
+def get_friends_unique_watched(user_data):
+    friend_movies = set()
+    user_movies = set()
+    unique_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.add(movie["title"])
+    for movie in user_data["watched"]:
+        user_movies.add(movie["title"])
+    unique_titles = friend_movies - user_movies
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] in unique_titles:
+                if movie not in unique_watched:
+                    unique_watched.append(movie)
+    return unique_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+# finds the movies the user hasn't watched, their friends have, AND they have access to (through their streaming service(s))
+def get_available_recs(user_data):
+    recs = []
+    friends_movies = get_friends_unique_watched(user_data)
+    for movie in friends_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            recs.append(movie)
+    return recs
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+# CONTEXT: my first attempt at finding the mode of the genre
+# def get_new_rec_by_genre(user_data):
+#     all_recs = get_available_recs(user_data)
+#     genre_dict = {}
+#     for movie in user_data["watched"]:
+#         for genre in movie["genre"]:
+#             if genre not in genre_dict:
+#                 genre_dict[genre] = 1
+#             else:
+#                 genre_dict[genre] += 1
+#     fav_genre = max(genre_dict)
+#         counter = 0
+#         # for genre in genre_dict:
+#         #     if genre_dict[genre] > 
+
+# gets recommendations for the user where:
+# they haven't watched it, their friends have, they have access to it, AND it's their most watched genre
+def get_new_rec_by_genre(user_data):
+    rec_by_genre = []
+    all_recs = get_available_recs(user_data)
+    genre_list = list()
+    try:
+        for movie in user_data["watched"]:
+            genre_list.append(movie["genre"])
+        fav_genre = max(set(genre_list), key = genre_list.count)
+        for movie in all_recs:
+            if movie["genre"] == fav_genre:
+                rec_by_genre.append(movie)
+        return rec_by_genre
+    except ValueError:
+        return []
+
+# finds a recommendation from the user to their friends where:
+# the movie is in their favorites list and none of the other friends have watched it
+def get_rec_from_favorites(user_data):
+    unique = get_unique_watched(user_data)
+    recs = []
+    for movie in user_data["favorites"]:
+        if movie in unique:
+            recs.append(movie)
+    return recs
+
+# the data below allowed me to run the debugger. 
+# if there is an easier way to do this, please let me know!!

```

Comment:
It's generally not a good practice to submit a lot of commented out code in a PR. There's a playground file which gives you access to all the data in `test_constants.py` so you shouldn't need to add additional data in this file.